idaes/core/base/unit_model.py

```diff
@@ -164,39 +162,24 @@ def add_port(blk, name=None, block=None, doc=None):
         Returns:
             A Pyomo Port object and associated components.
         """
-        # Validate block object
-        if not isinstance(block, StateBlock):
+        # Create Port
+        try:
+            port, ref_name_list = block.build_port(name, doc)
```

Comment:
We should probably do something to make sure name is not None here (or at least fail gracefully if it is).